### PR TITLE
Fix varmat tests to skip complex signatures

### PR DESCRIPTION
**Pull request by rok-cesnovar**
## Summary

This PR fixes varmat tests so that they now skip any signature with complex inputs.

## Checklist

- [x] Copyright holder: Rok Češnovar

    The copyright holder is typically you or your assignee, such as a university or company. By submitting this pull request, the copyright holder is agreeing to the license the submitted work under the following licenses:
      - Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
      - Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

**Review by WardBrian — APPROVED** [review]
Thanks!

**Review by SteveBronder — DISMISSED** [review]
Good!

**Comment by rok-cesnovar** [comment]
@SteveBronder @WardBrian I disabled the address sanitizer here, because this PR needs to be merged first in order to unblock everyone. 

As I stated in the other PR, this seems to be a weird issue that only pops up in the Docker container, no one has been able to reproduce it. We should add it back before the release or find a different way of running checks with the sanitizer.

Need one of you to re-approve in the next 8hours before the tests pass :)

**Review by WardBrian — APPROVED** [review]

**Comment by serban-nicusor-toptal** [comment]
```
==12319==LeakSanitizer has encountered a fatal error.

==12319==HINT: For debugging, try setting environment variable LSAN_OPTIONS=verbosity=1:log_threads=1

==12319==HINT: LeakSanitizer does not work under ptrace (strace, gdb, etc)
```
Based on this [stackoverflow thread](https://stackoverflow.com/questions/49735926/address-sanitizer-with-gcc-fails-on-ubuntu-17-10-docker-container) it looks like the container doesn't have the required privileges ( we refactored to avoid running as root or --privileged ).
I think that just adding `--cap-add SYS_PTRACE` in the Jenkinsfile should fix this error with Docker.
@rok-cesnovar can I test it in this PR or should I create another one after this one is merged?

**Comment by rok-cesnovar** [comment]
Lets merge this one to unblock everyone else and then debug without the added pressure.

**Pull request merged**
